lib/rules/no-useless-escape.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Look for useless escapes in strings and regexes
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var VALID_STRING_ESCAPES = {
+    "\\": true,
+    "n": true,
+    "r": true,
+    "v": true,
+    "t": true,
+    "b": true,
+    "f": true,
+    "u": true,
+    "x": true
+};
+
+var VALID_REGEX_ESCAPES = {
+    "\\": true,
+    "^": true,
+    "$": true,
+    "*": true,
+    "+": true,
+    "?": true,
+    ".": true,
+    "{": true,
+    "[": true,
+    "|": true,
+    "(": true,
+    "b": true,
+    "B": true,
+    "c": true,
+    "d": true,
+    "D": true,
+    "f": true,
+    "n": true,
+    "r": true,
+    "s": true,
+    "S": true,
+    "t": true,
+    "v": true,
+    "w": true,
+    "W": true,
+    "x": true,
+    "u": true
+};
+
+var nodeEscapes, nodeQuote;
+
+module.exports = function(context) {
+    /**
+     * Checks if the escape character in given slice is unnecessary.
+     *
+     * @private
+     * @param {string} elm - string slice to validate.
+     * @param {ASTNode} node - node to validate.
+     * @returns {void}
+     * @this escapes_quote_node
+     */
+    function validate(elm) {
+        if (!this.escapes[elm[1]] && !(this.quote && (elm[1] === this.quote))) {
+            context.report(this.node, "Unnecessary escape character: " + elm);
+        }
+    }
+
+    /**
+     * Checks if a node has an escape.
+     *
+     * @param {ASTNode} node - node to check.
+     * @returns {void}
+     */
+    function check(node) {
+        if (typeof node.value === "string") {
+            nodeEscapes = VALID_STRING_ESCAPES;
+            nodeQuote = node.raw[0];
+        } else if (node.regex) {
+            nodeEscapes = VALID_REGEX_ESCAPES;
+        } else {
+            return;
+        }
+        (node.raw.match(/\\[^\d]/g) || [])
+            .forEach(validate, {
```

Comment:
Putting this on the previous line may increase readability and save you an indent level.